src/fourierdomain/nuft/nuft.jl

```diff
@@ -8,21 +8,121 @@ Internal type used to store the cache for a non-uniform Fourier transform (NUFT)
+# Functions are overloaded based on the structure of RectiGrid
+function plan_indices(imgdomain::AbstractRectiGrid{<:Tuple{X,Y,Ti,Fr}},
+                      visdomain::UnstructuredDomain)
+    iminds = Tuple{Int,Int}[]
+    visinds = Vector{Int}[]
+    vis_points = domainpoints(visdomain)
+    Fr = imgdomain.Fr
+    Ti = imgdomain.Ti
+
+    for (j, fr) in pairs(Fr)
+        for (i, ti) in pairs(Ti)
+            push!(iminds, (i, j))
+            push!(visinds, findall(p -> p.Ti == ti && p.Fr == fr, vis_points))
+        end
+    end
+
+    return (iminds, visinds)
+end
+
+function plan_indices(imgdomain::AbstractRectiGrid{<:Tuple{X,Y,Fr,Ti}},
+                      visdomain::UnstructuredDomain)
+    iminds = Tuple{Int,Int}[]
+    visinds = Vector{Int}[]
+    vis_points = domainpoints(visdomain)
+    Ti = imgdomain.Ti
+    Fr = imgdomain.Fr
+
+    for (i, ti) in pairs(Ti)
+        for (j, fr) in pairs(Fr)
+            push!(iminds, (j, i))
+            push!(visinds, findall(p -> p.Ti == ti && p.Fr == fr, vis_points))
+        end
+    end
+    return (iminds, visinds)
+end
+
+function plan_indices(imgdomain::AbstractRectiGrid{<:Tuple{X,Y,Ti}},
+                      visdomain::UnstructuredDomain)
+    iminds = Int[]
+    visinds = Vector{Int}[]
+    vis_points = domainpoints(visdomain)
+    Ti = imgdomain.Ti
+
+    for (i, ti) in pairs(Ti)
+        push!(iminds, i)
+        push!(visinds, findall(p -> p.Ti == ti, vis_points))
+    end
+    return (iminds, visinds)
+end
+
+function plan_indices(imgdomain::AbstractRectiGrid{<:Tuple{X,Y,Fr}},
+                      visdomain::UnstructuredDomain)
+    iminds = Int[]
+    visinds = Vector{Int}[]
+    vis_points = domainpoints(visdomain)
+    Fr = imgdomain.Fr
+
+    for (j, fr) in pairs(Fr)
+        push!(iminds, j)
+        push!(visinds, findall(p -> p.Fr == fr, vis_points))
+    end
+    return (iminds, visinds)
+end
+
+# For only spatial case
+function plan_indices(imgdomain::AbstractRectiGrid{<:Tuple{X,Y}},
+                      visdomain::UnstructuredDomain)
+    return (0, 0)
+end
+
+# Function has been modified to process imgdomain/visdomain with Ti or Fr or both.
+# It calls the new plan_nuft  when Ti or Fr is present or else calls
+# the old spatial function
 function create_forward_plan(algorithm::NUFT, imgdomain::AbstractRectiGrid,
                              visdomain::UnstructuredDomain)
-    plan = plan_nuft(algorithm, imgdomain, visdomain)
     phases = make_phases(algorithm, imgdomain, visdomain)
-    return NUFTPlan(algorithm, plan, phases)
+    indices = plan_indices(imgdomain, visdomain)
+    if hasproperty(imgdomain, :Ti) || hasproperty(imgdomain, :Fr)
+        plan = plan_nuft(algorithm, imgdomain, visdomain, indices)
+    else
+        plan = plan_nuft_spatial(algorithm, imgdomain, visdomain)
+    end
+    return NUFTPlan(algorithm, plan, phases, indices, size(visdomain)[1])
 end
 
+# Added plan indices and totalvis points to the NUFTPlan in this function
```

Comment:
Again remove the description of what was changed from the code. This can go in the PR, but honestly it is pretty clear from the code what it does.